src/Pact/Types/Command.hs

```diff
@@ -157,9 +214,25 @@ keyPairsToSigners creds = map (uncurry keyPairToSigner) creds
 mkCommand' :: [(Ed25519KeyPair ,a)] -> ByteString -> IO (Command ByteString)
 mkCommand' creds env = do
   let hsh = hash env    -- hash associated with a Command, aka a Command's Request Key
-      toUserSig (cred,_) = signHash hsh cred
-  sigs <- traverse toUserSig creds
+      toUserSig (cred,_) = ED25519Sig $ signHash hsh cred
+  let sigs = toUserSig <$> creds
+  return $ Command env sigs hsh
+
+mkCommandWithDynKeys' :: [(DynKeyPair, a)] -> ByteString -> IO (Command ByteString)
+mkCommandWithDynKeys' creds env = do
+  let hsh = hash env    -- hash associated with a Command, aka a Command's Request Key
+  sigs <- traverse (toUserSig hsh) creds
   return $ Command env sigs hsh
+  where
+    toUserSig :: PactHash -> (DynKeyPair, a) -> IO UserSig
+    toUserSig hsh = \case
+      (DynEd25519KeyPair (pub, priv), _) ->
+        pure $ ED25519Sig $ signHash hsh (pub, priv)
+      (DynWebAuthnKeyPair _ pubWebAuthn privWebAuthn, _) -> do
+        signResult <- runExceptT $ signWebauthn pubWebAuthn privWebAuthn "" (toUntypedHash hsh)
+        case signResult of
+          Left e -> error e
```

Comment:
this error is pretty harsh

Reply:
(for now, I added some text to the error message to give more context if the user hits it)